File: app/src/main/java/com/trevorwiebe/trackacow/data/local/cacheDao/CacheFeedDao.java
package com.trevorwiebe.trackacow.data.local.cacheDao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.trevorwiebe.trackacow.data.cacheEntities.CacheFeedEntity;

import java.util.List;

@Dao
public interface CacheFeedDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertHoldingFeed(CacheFeedEntity cacheFeedEntity);

    @Insert
    void insertHoldingFeedList(List<CacheFeedEntity> holdingFeedEntities);

    @Query("SELECT * FROM holdingFeed")
    List<CacheFeedEntity> getHoldingFeedEntities();

    @Query("DELETE FROM holdingFeed")
    void deleteHoldingFeedTable();

    @Update
    void updateHoldingFeedEntity(CacheFeedEntity cacheFeedEntity);

    @Delete
    void deleteFeedEntity(CacheFeedEntity cacheFeedEntity);
}

File: app/src/main/java/com/trevorwiebe/trackacow/data/local/cacheDao/CacheFeedDao.kt
package com.trevorwiebe.trackacow.data.local.cacheDao

import androidx.room.*
import com.trevorwiebe.trackacow.data.cacheEntities.CacheFeedEntity

@Dao
interface CacheFeedDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertHoldingFeedList(holdingFeedEntities: List<CacheFeedEntity>)

    @Update
    suspend fun updateHoldingFeedEntity(cacheFeedEntity: CacheFeedEntity)

    @Query("SELECT * FROM cache_feed")
    suspend fun getCacheFeeds(): List<CacheFeedEntity>

    @Delete
    suspend fun deleteFeedEntity(cacheFeedEntity: CacheFeedEntity)

}